pkgs/test_core/lib/src/runner/configuration.dart

```diff
@@ -73,6 +73,15 @@ class Configuration {
   String get reporter => _reporter ?? defaultReporter;
   final String _reporter;
 
+  /// The map of file reporters where the key is the name of the reporter and
+  /// the value is the filepath to which its output should be written.
+  final Map<String, String> fileReporters;
```

Comment:
Realistically right now there would only ever be one json file reporter entry, but if there are additional reporters added/supported in the future, this would support multiple output formats (e.g. json and xunit).